lib/src/request.dart

```diff
@@ -130,6 +133,28 @@ class Request extends Message {
       {Map<String, String> headers, Map<String, Object> context})
       : this('DELETE', url, headers: headers, context: context);
 
+  factory Request.multipart(url,
+      {Map<String, String> headers,
+        Map<String, Object> context,
+        Map<String, String> fields,
+        Iterable<MultipartFile> files}) {
+    fields ??= <String, String>{};
+    files ??= <MultipartFile>[];
+    headers ??= <String, String>{};
+
+    var boundary = _boundaryString();
+
+    return new Request._(
+        'POST',
```

Comment:
Technically, multipart requests don't always have to be POST. They are the vast majority of the time, so it's a good default, but we should probably add an optional `method` parameter.